main.tf

```diff
@@ -112,67 +97,47 @@ resource "aws_route_table" "public-RTB" {
   }
 
   tags = {
-      Name = "public-RTB"
+    Name = "public-RTB"
   }
 }
 
 #private-aws_route_table with nat gateway
-resource "aws_route_table" "private-1-RTB" {
-  vpc_id = aws_vpc.main.id
-
-  route {
-    cidr_block = "0.0.0.0/0"
-    nat_gateway_id = aws_nat_gateway.public-subnet-1.id
-  }
-
-  tags = {
-      Name = "private-RTB-1"
-  }
-}
+resource "aws_route_table" "private-RTB" {
 
-
-#private-aws_route_table with nat gateway
-resource "aws_route_table" "private-2-RTB" {
+  count  = 2
   vpc_id = aws_vpc.main.id
 
   route {
-    cidr_block = "0.0.0.0/0"
-    nat_gateway_id = aws_nat_gateway.public-subnet-2.id
+    cidr_block     = "0.0.0.0/0"
+    nat_gateway_id = aws_nat_gateway.nat[count.index].id
   }
 
   tags = {
-      Name = "private-RTB-2"
+    Name = "private-RTB${count.index + 1}"
   }
 }
 
-#aws_route_table_association-public-subnet-1-public-RTB
-resource "aws_route_table_association" "public-subnet-1" {
-  subnet_id      = aws_subnet.public-subnet-1.id
-  route_table_id = aws_route_table.public-RTB.id
-}
 
+#aws_route_table_association-public-subnets
+resource "aws_route_table_association" "public" {
 
-#aws-aws_route_table_association-public-subnet-2-public-RTB
-resource "aws_route_table_association" "public-subnet-2" {
-  subnet_id      = aws_subnet.public-subnet-2.id
+  count          = 2
+  subnet_id      = aws_subnet.public[count.index]
```

Comment:
rewrite `subnet_id      = aws_subnet.public[count.index]` to `subnet_id      = aws_subnet.public[count.index].id`